Paintroid/src/main/java/org/catrobat/paintroid/dialog/SaveInformationDialog.kt

```diff
@@ -0,0 +1,192 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.dialog
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.graphics.Bitmap
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemSelectedListener
+import android.widget.ArrayAdapter
+import android.widget.ImageButton
+import android.widget.SeekBar
+import android.widget.SeekBar.OnSeekBarChangeListener
+import android.widget.Spinner
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import org.catrobat.paintroid.FileIO
+import org.catrobat.paintroid.R
+import org.catrobat.paintroid.common.Constants.IS_NO_FILE
+import org.catrobat.paintroid.common.MainActivityConstants.PERMISSION_EXTERNAL_STORAGE_SAVE_COPY
+
+class SaveInformationDialog(
+    permissionCode: Int,
+    imageNumber: Int,
+    isStandard: Boolean,
+    isExport: Boolean
+) :
+    MainActivityDialogFragment(),
+    OnItemSelectedListener,
+    OnSeekBarChangeListener {
+
+    private lateinit var mySpinner: Spinner
+    private lateinit var inflater: LayoutInflater
+    private lateinit var specificFormatLayout: ViewGroup
+    private lateinit var jpgView: View
+    private lateinit var mySeekbar: SeekBar
+    private lateinit var percentage: TextView
+    private lateinit var informationButton: ImageButton
+    private lateinit var imageName: TextView
+    private lateinit var setName: String
+    private var permission = 0
+    private var isExport = false
+
+    init {
+        if (isStandard) {
+            FileIO.filename = "image"
+            setFileDetails(Bitmap.CompressFormat.PNG, false, ".png")
+        }
+        val bundle = Bundle().apply {
+            if (FileIO.filename == "image") {
+                putString("setName", FileIO.filename + imageNumber)
+            } else {
+                putString("setName", FileIO.filename)
+            }
+            putInt("permission", permissionCode)
+            putBoolean("isExport", isExport)
+        }
+        arguments = bundle
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val arguments = requireArguments()
+        permission = arguments.getInt("permission")
+        setName = arguments.getString("setName").toString()
+        isExport = arguments.getBoolean("isExport")
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initializeViews(view)
+        initializeFunctioning()
+        when {
+            FileIO.isCatrobatImage -> mySpinner.setSelection(2)
+            FileIO.compressFormat == Bitmap.CompressFormat.PNG -> mySpinner.setSelection(0)
+            else -> mySpinner.setSelection(1)
+        }
+    }
```

Comment:
```suggestion
        super.onViewCreated(view, savedInstanceState)
        initializeViews(view)
        initializeFunctioning()
        initFoo()
    }
```
Please try to keep the same level of abstraction ;)

Reply:
done